src/components/Vote.jsx

```diff
@@ -0,0 +1,100 @@
+import { useState, useEffect, useRef } from "react";
+import { patchVote } from "../utils";
+
+const Vote = ({ review }) => {
+  const initialRenderTracker = useRef(true);
+  const [voteValue, setVoteValue] = useState({
+    vote: 0,
+    amount: 0,
+  });
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const upvote = () => {
+    switch (voteValue.vote) {
+      case -1:
+        setVoteValue({ vote: 1, amount: 2 });
+        break;
+      case 0:
+        setVoteValue({ vote: 1, amount: 1 });
+        break;
+      case 1:
+        setVoteValue({ vote: 0, amount: -1 });
+        break;
+    }
+  };
+
+  const downvote = () => {
+    switch (voteValue.vote) {
+      case -1:
+        setVoteValue({ vote: 0, amount: 1 });
+        break;
+      case 0:
+        setVoteValue({ vote: -1, amount: -1 });
+        break;
+      case 1:
+        setVoteValue({ vote: -1, amount: -2 });
+        break;
+    }
+  };
+
+  useEffect(() => {
+    if (initialRenderTracker.current) {
```

Comment:
I'm interested in what this is for? Why do you want to track if it's the first render or not?